cfpb/consumerfinance.gov
# Order prepaid agreements based on effective_date

Previously, prepaid agreements were ordered based on their created_time, so the agreement with the latest created_time would be designated "most recent" in the UI. Now, since the intake process for these agreements has improved and the effective date field is more reliable, the stakeholder has asked to use effective_date to order the agreements.

Now, for a prepaid product, the "most recent" agreement will be the one whose effective date is latest. If multiple agreements share the latest effective date, the one with the latest created_time will be designated "most recent".


## Changes

- change ordering of prepaid agreements, update tests accordingly


## How to test this PR

Run this branch. Check that prepaid agreements are ordered by effective date. 

Here are some examples of prepaid products where this update changes the designated "most recent" agreement vs what's in Production:

- Brightwell Visa Prepaid Card (Holland) - [Production](https://www.consumerfinance.gov/data-research/prepaid-accounts/search-agreements/detail/46085/) vs [local](http://localhost:8000/data-research/prepaid-accounts/search-agreements/detail/46085/)
- OceanPay Visa Prepaid Card (Carnival) - [Production](https://www.consumerfinance.gov/data-research/prepaid-accounts/search-agreements/detail/46057/) vs [local](http://localhost:8000/data-research/prepaid-accounts/search-agreements/detail/46057/)
- Government Payments - [Production](https://www.consumerfinance.gov/data-research/prepaid-accounts/search-agreements/detail/43170/) vs [local](http://localhost:8000/data-research/prepaid-accounts/search-agreements/detail/43170/)


## Notes and todos

- The agreement intake team will also be making the same change on their side. We want our change and theirs to go live at about the same time, to reduce potential for confusion.


## Checklist



  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance
